app/Commands/UploadTrackCommand.php

```diff
@@ -40,6 +39,21 @@ class UploadTrackCommand extends CommandBase
     private $_allowShortTrack;
     private $_customTrackSource;
     private $_autoPublishByDefault;
+    private $_track;
+    private $_version;
+    private $_isReplacingTrack;
+
+    /**
+     * @return bool
+     */
+    public function authorize()
+    {
+        if ($this->_isReplacingTrack) {
+            return $this->_track && Gate::allows('edit', $this->_track);
```

Comment:
Checking whether the given track exists is unnecessary if you declare the type of the constructor's `$track` parameter.